src/Promitor.Agents.Scraper/Validation/Steps/ResourceDiscoveryValidationStep.cs

```diff
@@ -4,28 +4,38 @@
 using Microsoft.Extensions.Logging.Abstractions;
 using Promitor.Agents.Scraper.Configuration;
 using Promitor.Agents.Scraper.Validation.Interfaces;
+using Promitor.Core.Scraping.Configuration.Providers.Interfaces;
+using Promitor.Core.Scraping.Configuration.Serialization;
 
 namespace Promitor.Agents.Scraper.Validation.Steps
 {
     public class ResourceDiscoveryValidationStep : ValidationStep, IValidationStep
     {
+        private readonly IMetricsDeclarationProvider _metricsDeclarationProvider;
         private readonly ResourceDiscoveryConfiguration _configuration;
 
-        public ResourceDiscoveryValidationStep(ResourceDiscoveryConfiguration configuration) : this(configuration, NullLogger.Instance)
+        public ResourceDiscoveryValidationStep(ResourceDiscoveryConfiguration configuration, IMetricsDeclarationProvider metricsDeclarationProvider) : this(configuration, metricsDeclarationProvider, NullLogger.Instance)
         {
         }
 
-        public ResourceDiscoveryValidationStep(ResourceDiscoveryConfiguration configuration, ILogger logger) : base( logger)
+        public ResourceDiscoveryValidationStep(ResourceDiscoveryConfiguration configuration, IMetricsDeclarationProvider metricsDeclarationProvider, ILogger logger) : base( logger)
         {
+            _metricsDeclarationProvider = metricsDeclarationProvider;
             _configuration = configuration;
         }
 
         public string ComponentName { get; } = "Resource Discovery";
 
         public ValidationResult Run()
         {
+            var doesDeclareResourceDiscoveryGroups = DetermineIfDiscoveryGroupsAreDefined();
             if (_configuration == null)
             {
+                if (doesDeclareResourceDiscoveryGroups)
+                {
+                    return ValidationResult.Failure(ComponentName, new List<string> {"No resource discovery is configured while discovery groups are defined to scrape"});
```

Comment:
Can I maybe suggest rewording this to something like:

```
Resource discovery groups are defined in your metrics configuration, but resource discovery has not been configured. Please add a resource discovery configuration.
```

I just feel like it's a bit more explicit then, and it also points people towards a solution.

Reply:
Sure, good suggestion!